test/extended/node_tuning/node_tuning_utils.go

```diff
@@ -0,0 +1,192 @@
+	}
+
+	conditions, found, err := unstructured.NestedFieldNoCopy(pool.Object, "status", "conditions")
+	if err != nil || !found {
+		return false, false
+	}
+	original, ok := conditions.([]interface{})
+	if !ok {
+		return false, false
+	}
+	var updated, updating, degraded bool
+	for _, obj := range original {
+		o, ok := obj.(map[string]interface{})
+		if !ok {
+			return false, false
+		}
+		t, found, err := unstructured.NestedString(o, "type")
+		if err != nil || !found {
+			return false, false
+		}
+		s, found, err := unstructured.NestedString(o, "status")
+		if err != nil || !found {
+			return false, false
+		}
+		if t == "Updated" && s == "True" {
+			updated = true
+		}
+		if t == "Updating" && s == "True" {
+			updating = true
+		}
+		if t == "Degraded" && s == "True" {
+			degraded = true
+		}
+	}
+	if paused {
+		framework.Logf("Pool %s is paused, treating as up-to-date (Updated: %v, Updating: %v, Degraded: %v)", name, updated, updating, degraded)
+		return true, updating
+	}
+	if updated && !updating && !degraded {
+		return true, updating
+	}
+	framework.Logf("Pool %s is still reporting (Updated: %v, Updating: %v, Degraded: %v)", name, updated, updating, degraded)
+	return false, updating
+}
+
+func findCondition(conditions []configv1.ClusterOperatorStatusCondition, name configv1.ClusterStatusConditionType) *configv1.ClusterOperatorStatusCondition {
+	for i := range conditions {
+		if name == conditions[i].Type {
+			return &conditions[i]
+		}
+	}
+	return nil
+}
+
+func assertCOStatus(oc *exutil.CLI, coName string) {
+	var (
+		clusterOperators []configv1.ClusterOperator
+		desiredCO        configv1.ClusterOperator
+	)
+	clusterOperatorsList, err := oc.AdminConfigClient().ConfigV1().ClusterOperators().List(context.Background(), metav1.ListOptions{})
+	o.Expect(err).ToNot(o.HaveOccurred())
+	clusterOperators = clusterOperatorsList.Items
+	for _, clusterOperator := range clusterOperators {
+		if clusterOperator.Name == coName {
+			desiredCO = clusterOperator
+			e2e.Logf("desiredCO Name is %v", desiredCO.Name)
+			e2e.Logf("desiredCO.status.conditions of %v is %v", desiredCO.Name, desiredCO.Status.Conditions)
+			break
+		}
+	}
+	errWait := wait.Poll(30*time.Second, 600*time.Second, func() (bool, error) {
+		available := findCondition(desiredCO.Status.Conditions, configv1.OperatorAvailable)
+		degraded := findCondition(desiredCO.Status.Conditions, configv1.OperatorDegraded)
+		progressing := findCondition(desiredCO.Status.Conditions, configv1.OperatorProgressing)
+		if available.Status == configv1.ConditionTrue &&
+			degraded.Status == configv1.ConditionFalse &&
+			progressing.Status == configv1.ConditionFalse {
+			e2e.Logf("the status of status of cluster operator %v restore to normal state", coName)
+			return true, nil
+		}
+		e2e.Logf("the status of machine-config co is: available.Status [%v]  degraded.Status [%v] and progressing.Status [%v] , check again", available.Status, degraded.Status, progressing.Status)
```

Comment:
s/] , check/], check/
s/  degraded/ degraded/

Reply:
Fixed in new code